gqcpy/include/interfaces.hpp

```diff
@@ -699,4 +700,145 @@ void bindSimple2DMInterface(Class& py_class) {
 }
 
 
+/*
+ *  MARK: QCModel - Hartree-Fock
+ */
+
+/**
+ *  Add Python bindings for some APIs related to `QCModel::R/U/GHF`.
+ * 
+ *  @tparam Class               The type of the Pybind11 `class_` (generated by the compiler).
+ *  
+ *  @param py_class             The Pybind11 `class_` that should obtain APIs related to `QCModel::R/U/GHF`.
+ */
+template <typename Class>
+void bindQCModelHartreeFockInterface(Class& py_class) {
+
+    // The C++ type corresponding to the Python class.
+    using Type = typename Class::type;
+    using Hamiltonian = typename Type::Hamiltonian;
+    using Scalar = typename Type::Scalar;
+
+    py_class
+        .def(
+            "calculateOrthonormalBasis1DM",
+            [](const Type& parameters) {
+                return parameters.calculateOrthonormalBasis1DM();
+            },
+            "Return the 1-DM expressed in an orthonormal basis related to these optimal HF parameters.")
+
+        .def(
+            "calculateScalarBasis1DM",
+            [](const Type& parameters) {
+                return parameters.calculateScalarBasis1DM();
+            },
+            "Return the HF 1-DM in the scalar/AO basis related to these optimal parameters.")
+
+        .def(
+            "calculateStabilityMatrices",
+            [](const Type& parameters, const Hamiltonian& hamiltonian) {
+                return parameters.calculateStabilityMatrices(hamiltonian);
+            },
+            "Return the HF stability matrices parameters.")
+
+        .def("expansion",
+             &Type::expansion,
+             "Return the coefficient matrix that expresses every spatial orbital (as a column) in its underlying scalar basis.")
+
+        .def(
+            "numberOfElectrons",
+            [](const Type& parameters) {
+                return parameters.numberOfElectrons();
+            },
```

Comment:
```suggestion
            &Type::numberOfElectrons,
```

Reply:
Writing it this way will give an error as pybind won't be able to recognise which of the overloaded `numberOfElectrons()` API's of the RHF model it is supposed to use.

Reply:
Yep, an oversight on my part.